### PR TITLE
Fixed #14135 `yii\web\Request::getBodyParam()` crashes on object body

### LINKED ISSUE #14135: \yii\web\Request::getBodyParam crashes when \yii\web\JsonParser is configured to return objects
When the `JsonParser` is configured to return objects via `asArray = false` `Request::getBodyParam()` will crash as it expects to get an array from the `getBodyParams`. By returning an object the `JsonParser` also breaks its own interface. But returning an object could be quite handy sometimes, so I think that the `\yii\web\RequestParserInterface::parse` interface should be extended to incorporate this. Thoughts?

| Yii version      | 2.0.11.2

### PR DESCRIPTION
**Pull request by klimov-paul**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14135

### CODE REVIEW

File: framework/web/Request.php [hunk 1]
```diff
@@ -582,6 +582,15 @@ public function getBodyParam($name, $defaultValue = null)
     {
         $params = $this->getBodyParams();
 
+        if (is_object($params)) {
+            // unable to use `ArrayHelper::getValue()` due to different dots in key logic and lack of exception handling
+            try {
+                return $params->{$name};
+            } catch (\Exception $e) {
+                return $defaultValue;
```
Comment (samdark) [inline framework/web/Request.php]: Is that expected behavior?

Comment (klimov-paul) [reply framework/web/Request.php]: See https://github.com/yiisoft/yii2/blob/master/framework/helpers/BaseArrayHelper.php#L206-L209

I can not find any other way to handle unexisting object property here.

Comment (samdark) [reply framework/web/Request.php]: OK.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**